source/slang/slang-check-decl.cpp

```diff
@@ -915,6 +916,87 @@ namespace Slang
             //
             validateArraySizeForVariable(varDecl);
         }
+
+        // The NVAPI library allows user code to express extended operations
+        // (not supported natively by D3D HLSL) by communicating with
+        // a specially identified shader parameter called `g_NvidiaExt`.
+        //
+        // By default, that shader parameter would look like an ordinary
+        // global shader parameter to Slang, but we want to be able to
+        // associate special behavior with it to make downstream compilation
+        // work nicely (especially in the case where certain cross-platform
+        // operations in the Slang standard library need to use NVAPI).
+        //
+        // We will detect a global variable declaration that appears to
+        // be declaring `g_NvidiaExt` from NVAPI, and mark it with a special
+        // modifier to allow downstream steps to detect it whether or
+        // not it has an associated name.
+        //
+        if( as<ModuleDecl>(varDecl->parentDecl)
```

Comment:
It may be worth holding g_NvidiaExt (odd capitalization...) as a Name for faster comparison.